File: VB/Controllers/HomeController.vb
Imports System.Web.Mvc

Namespace GridUpdateComboMvc
	Public Class HomeController
		Inherits Controller

		Public Function Index() As ActionResult
			Return View()
		End Function

		Public Function SimplestApproach() As ActionResult
			Return View()
		End Function

		Public Function ServerSideCalculation() As ActionResult
			Return View()
		End Function

		Public Function HiddenColumn() As ActionResult
			Return View()
		End Function

		Public Function GridViewPartial(ByVal hiddenColumnScenario? As Boolean) As ActionResult
			ViewData("hiddenColumnScenario") = hiddenColumnScenario
			Return PartialView()
		End Function

		Public Function GridViewUpdatePartial(ByVal product As Product) As ActionResult
			' Update logic is not implemented in this example
			Return PartialView("GridViewPartial")
		End Function

		Public Function ServerSideCalculationCallbackActionMethod(ByVal categoryID As Integer) As ActionResult
			Dim someConstant As Decimal = 10D
			Dim result As String = (categoryID * someConstant).ToString()

			Return Content(result)
		End Function

		Public Function HiddenColumnCallbackActionMethod(ByVal categoryID As Integer) As ActionResult
			Dim someConstant As String = " Test"
			Dim description As String = NorthwindDataProvider.GetCategoryDescriptionById(categoryID)

			Return Content(description & someConstant)
		End Function
	End Class
End Namespace

File: CS/Controllers/HomeController.cs
using System.Web.Mvc;

namespace GridUpdateComboMvc {
    public class HomeController : Controller {
        public ActionResult Index() {
            return View();
        }

        public ActionResult SimplestApproach() {
            return View();
        }

        public ActionResult ServerSideCalculation() {
            return View();
        }

        public ActionResult HiddenColumn() {
            return View();
        }

        public ActionResult GridViewPartial(bool? hiddenColumnScenario) {
            ViewData["hiddenColumnScenario"] = hiddenColumnScenario;
            return PartialView();
        }

        public ActionResult GridViewUpdatePartial(Product product) {
            // Update logic is not implemented in this example
            return PartialView("GridViewPartial");
        }

        public ActionResult ServerSideCalculationCallbackActionMethod(int categoryID) {
            decimal someConstant = 10m;
            string result = (categoryID * someConstant).ToString();

            return Content(result);
        }

        public ActionResult HiddenColumnCallbackActionMethod(int categoryID) {
            string someConstant = " Test";
            string description = NorthwindDataProvider.GetCategoryDescriptionById(categoryID);

            return Content(description + someConstant);
        }
    }
}